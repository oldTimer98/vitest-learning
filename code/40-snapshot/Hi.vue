<template>
	<div id="hi">
		hi how are you
	</div>
</template>
