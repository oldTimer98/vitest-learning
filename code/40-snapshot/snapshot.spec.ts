import { it, expect, describe } from 'vitest'
import Hi from './Hi.vue';
import {mount} from '@vue/test-utils';




it('snapshot', () => {
	// form rules
	const wrapper = mount(Hi)
	expect(wrapper.html()).toMatchSnapshot()
	// expect({name:"cxr", age: 10}).toMatchSnapshot()
});


it('inline', () => {
	
	expect({name:"cxr", age: 10}).toMatchInlineSnapshot(`
		{
		  "age": 10,
		  "name": "cxr",
		}
	`)
	
});


it('file', () => {
	const wrapper = mount(Hi)
	expect(wrapper.html()).toMatchFileSnapshot("./hi.html")
	
});

