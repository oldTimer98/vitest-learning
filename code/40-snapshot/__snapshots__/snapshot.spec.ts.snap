// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`file 1`] = `"<div id=\\"hi\\"> hi </div>"`;

exports[`snapshot 1`] = `"<div id=\\"hi\\"> hi how are you </div>"`;
